index.js

```diff
@@ -11,50 +11,101 @@ var couch2pg = require('./libs/couch2pg/importer')(
       db, couchdb,
       env.couch2pgDocLimit,
       env.couch2pgChangesLimit),
-    xmlforms = require('./libs/xmlforms/extractor')(db);
-
-var batchLoop = function() {
-  var completeBatch = function(lastSeq) {
-    return couch2pg.storeSeq(lastSeq)
-      .then(function() {
-        log.info('Batch completed up to ' + lastSeq);
-      });
-  };
-
-  log.info('Beginning couch2pg and xmlforms batch');
-
-  return couch2pg.importBatch()
-  .then(function(changes) {
-    var allDocs = changes.deleted.concat(changes.edited);
-    log.debug('There are ' + allDocs.length + ' changes');
-
-    if (allDocs.length > 0) {
-      return xmlforms.extract(allDocs)
-      .then(function() {
-        return completeBatch(changes.lastSeq);
-      })
-      .then(batchLoop);
+    xmlforms = require('./libs/xmlforms/updater')(db);
+
+var fallback = 0;
+var sleepMs = function(errored) {
+  if (errored) {
+    var fallbackMs = fallback * 1000 * 60;
+    if (fallbackMs < env.sleepMs) {
+      fallback++;
+      return fallbackMs;
+    } else {
+      return env.sleepMs;
+    }
+  } else {
+    fallback = 0;
+    return env.sleepMs;
+  }
+};
+
+var migrateCouch2pg = function() {
+  return couch2pgMigrator(env.postgresqlUrl)();
+};
+var migrateXmlforms = function() {
+  return xmlformsMigrator(env.postgresqlUrl)();
+};
+
+var delayLoop = function(loopFn, errored) {
+  return new Promise(function(resolve) {
+    var ms = sleepMs(errored);
+    log.info('Run '+(errored ? 'errored' : 'complete') + '. Next run at ' + new Date(new Date().getTime() + ms));
+    if (ms === 0) {
```

Comment:
Ok I see what you're saying now. I think on my first read through I misread one line, so ignore that comment.

The good news however is reading through it again I found another issue... the loopFn is never used. Maybe this is on purpose, but it still makes it confusing. My assumption was that the loopFn would be retried until it passed, but that's not what this does currently, right?

Secondly, I think you should use setTimeout not setInterval - the latter will attempt to resolve the promise multiple times.

It'd be good to chat through this line by line to make sure I'm understanding what's meant to be going on.